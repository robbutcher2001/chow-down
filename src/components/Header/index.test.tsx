import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import Header from '.';

test('Header basic snapshot render', () => {
  const header = renderer.create(
    <Router history={createMemoryHistory()}>
      <Header />
    </Router>
  );

  expect(header.toJSON()).toMatchSnapshot();
});

test('Header correct tag content assertion', () => {
  const { getByText } = render(
    <Router history={createMemoryHistory()}>
      <Header />
    </Router>
  );

  expect(getByText(/Chow Down/).textContent).toEqual('Chow Down');
  expect(getByText(/Chow down on/).textContent).toEqual('Chow down on a weekly plan of your evening meals.');
});