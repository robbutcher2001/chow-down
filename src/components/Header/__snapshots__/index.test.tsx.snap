// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header basic snapshot render 1`] = `
<header
  className="sc-bdVaJa gEKKMn"
>
  <h1>
    <a
      href="/"
      onClick={[Function]}
    >
      Chow Down
    </a>
  </h1>
  <p>
    Chow down on a weekly plan of your evening meals.
  </p>
</header>
`;
